csrc/host_ir/lower.cpp

```diff
@@ -304,6 +304,10 @@ std::vector<Expr*> HostIrLower::lower(Expr* c) {
       lowerToBroadcastOrSendRecv(input_tv, output_tv, comms);
     }
   }
+
+  std::for_each(comms.begin(), comms.end(), [this](Expr* comm) {
+    comm->as<Communication>()->backend() = params_.communicator_backend;
```

Comment:
Can we pass communicator_backend into lowerTo* so it can simplify construct Communications with the right backend type? Alternatively, we can make lowerTo* instance methods so it can access params_. 

In general, I'd avoid exposing a field as a mutable reference. It allows anyone to update the field at any time, making code harder to reason about.